packages/java/src/browser/java-client-contribution.ts

```diff
@@ -121,7 +125,29 @@ export class JavaClientContribution extends BaseLanguageClientContribution {
 
     protected getStartParameters(): MaybePromise<JavaStartParams> {
         const workspace = this.workspace.rootUri ? this.workspace.rootUri : undefined;
-        return { workspace };
+        const jvmArgs: string[] = [];
+        const vmargsLine = this.preferenceService.get('java.jdt.ls.vmargs', '');
+        this.parseVMargs(jvmArgs, vmargsLine);
+        return { workspace, jvmArgs};
     }
 
+    // copied from https://github.com/redhat-developer/vscode-java/blob/master/src/javaServerStarter.ts#L105
```

Comment:
@marcdumais-work do we need a CQ for this copied code?

Reply:
@vince-fugnitto @marcdumais-work 
I referred to the following.
https://github.com/theia-ide/theia/blob/master/packages/java/src/browser/java-preferences.ts#L19

Reply:
Yes, I think we need a CQ. We had at least one before for that project, but as far as I can tell, not that specific file. 

@daisuke-yoshimoto can you confirm if you copied the `parseVMArgs` function below from the master branch of the `redhat-developer/vscode-java` project?

Reply:
@marcdumais-work 
Yes. However, I made a few changes to fix lint errors.
```
arg = arg.replace(/(\\)?"/g, ($0, $1) => { return ($1 ? $0 : ''); });
```
```
arg = arg.replace(/(\\)?"/g, ($0, $1) => ($1 ? $0 : ''));
```

Reply:
Thanks for the info @daisuke-yoshimoto . If we want to keep this copied code we need to register a CQ (Contribution Questionaire) with the Eclipse Foundation, so they can validate that the licenses are compatible, etc... 

I'll go ahead and register the CQ. It can take from a couple of days to a few weeks before approval, and we'll need to wait before we merge this contribution.

Reply:
@marcdumais-work 
Thanks a lot 😄

Reply:
CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19687  (Eclipse account required to access)

Reply:
@daisuke-yoshimoto I think we should move this line ("// copied from https://github.com/redhat-developer/vscode-java/blob/master/src/javaServerStarter.ts#L105") to the top of the file, just under the original header and also mention the license of the copied code. Something like this:

https://github.com/theia-ide/theia/blob/master/packages/java/src/browser/java-preferences.ts#L17-L19

However the link should be the one that will map to the specific version of the file we copied. I think this is ok: https://github.com/redhat-developer/vscode-java/blob/v0.44.0/src/javaServerStarter.ts#L105-L122

Reply:
@marcdumais-work 
I fixed. Please check again.
https://github.com/theia-ide/theia/pull/5074/files#diff-b16869c7f6a204b1614c1e58569863cbR17